test/es-module/test-esm-import-flag.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+require('../common');
+const fixtures = require('../common/fixtures');
+const assert = require('assert');
+const { spawnSync } = require('child_process');
+
+{
+  // known limitation:
+  // imports should not be imported when evaluating text
+  const { status, stderr, stdout } = spawnSync(
+    process.execPath,
+    [
+      '--import',
+      fixtures.path('es-modules', 'mjs-file.mjs'),
+      '-e', 'console.log("log")',
+    ],
+    { encoding: 'utf8' },
+  );
+
+  assert.strictEqual(status, 0);
+  assert.strictEqual(stderr, '');
+  assert.strictEqual(stdout, 'log\n');
+}
+
+{
+  const { status, stderr, stdout } = spawnSync(
+    process.execPath,
+    [
+      '--import',
+      fixtures.path('es-modules', 'mjs-file.mjs'),
+      fixtures.path('es-modules', 'cjs-file.cjs'),
+    ],
+    { encoding: 'utf8' },
+  );
+
+  assert.strictEqual(status, 0);
+  assert.strictEqual(stderr, '');
+  assert.strictEqual(stdout, '.mjs file\n.cjs file\n');
+}
+
+{
+  const { status, stderr, stdout } = spawnSync(
+    process.execPath,
+    [
+      '--import',
+      fixtures.path('es-modules', 'mjs-file.mjs'),
+      fixtures.path('es-modules', 'mjs-file.mjs'),
+    ],
+    { encoding: 'utf8' },
+  );
+
+  assert.strictEqual(status, 0);
+  assert.strictEqual(stderr, '');
+  assert.strictEqual(stdout, '.mjs file\n.mjs file\n');
+}
+
+{
+  const start = Date.now();
+  const { status, stderr, stdout } = spawnSync(
+    process.execPath,
+    [
+      '--import',
+      fixtures.path('es-modules', 'esm-top-level-await.mjs'),
+      fixtures.path('es-modules', 'print_imported.mjs'),
+    ],
+    { encoding: 'utf8' },
+  );
```

Comment:
All of these tests that have no entry point are unrealistic. Can we add an entry point file to each of them, say something that prints something, and check for that in the output? In particular, for this test what’s the order of what’s printed between the `--import` modules and the entry point?

Reply:
`print_imported.mjs` is the entry point. am I missing anything?

Reply:
Ah sorry I missed that somehow. Then can we have a test that an `--import` module that includes top-level `await`, then prints something, has its output printed before any user code that also prints something?

Reply:
I tested the timestamps align at order, is that ok?

Reply:
I guess I’m just not following all that’s going on in that file, or why it’s so complicated. I had something like this in mind:

```js
// esm-top-level-await.mjs
await new Promise(resolve => setTimeout(() => {
  console.log(1);
  resolve();
}, 10));

console.log(2);

// entry.mjs
console.log('entry');
```

And the test would confirm that `1` is printed before `2` is printed before `entry`.

As a follow up to this, what would we expect (cc @aduh95 @JakobJingleheimer) the behavior to be if the first file included this?

```js
setTimeout(() => {
  console.log(3);
}, 1000);
```

Since it’s not awaited, does this `3` get printed before the entrypoint’s `entry`?

Reply:
fixed the test,
I assume if not awaited it should not wait for it - for example, if some `--import` runs some APM keep alive in an interval or in a timeot, I would expect the program to start regardless of that